src/rtc/rtc_s1_single_job.py

```diff
@@ -373,6 +375,103 @@ def save_browse(imagery_list, browse_image_filename,
     logger.info(f'file saved: {browse_image_filename}')
 
 
+def save_browse_static(filename, browse_image_filename,
+                       browse_image_height, browse_image_width,
+                       temp_files_list, scratch_dir, logger):
+    """Create and save a browse image for the RTC-S1-STATIC product
+
+       Parameters
+       ----------
+       filename : str
+           Static layer filename
+       browse_image_filename : str
+           Output browse file
+       browse_image_height : int
+           Browse image height
+       browse_image_width : int
+           Browse image width
+       scratch_dir : str
+           Directory for temporary files
+       temp_files_list: list (optional)
+           Mutable list of temporary files. If provided,
+           paths to the temporary files generated will be
+           appended to this list.
+       logger : loggin.Logger
+           Logger
+    """
+
+    BROWSE_IMAGE_MIN_PERCENTILE = 3
+    BROWSE_IMAGE_MAX_PERCENTILE = 97
+
+    logger.info(f'creating browse image: {browse_image_filename}')
+    logger.info(f'            from file: {filename}')
+
+    alpha_channel = None
+
+    gdal_ds = gdal.Open(filename, gdal.GA_ReadOnly)
+    image_width = gdal_ds.GetRasterBand(1).XSize
+    image_height = gdal_ds.GetRasterBand(1).YSize
+
+    if (browse_image_height is not None or
+            browse_image_width is not None):
+
+        del gdal_ds
+
+        if browse_image_width is None:
+            browse_image_width = int(np.round(
+                (browse_image_height * float(image_width) / image_height)))
+
+        if browse_image_height is None:
+            browse_image_height = int(np.round(
+                (browse_image_width * float(image_height) / image_width)))
+
+        logger.info(f'        browse length: {browse_image_height}')
+        logger.info(f'        browse width: {browse_image_width}')
+
+        browse_temp_file = tempfile.NamedTemporaryFile(
+            dir=scratch_dir, suffix='.tif').name
+
+        if temp_files_list is not None:
+            temp_files_list.append(browse_temp_file)
+
+        resamp_algorithm = 'CUBIC'
+
+        # Translate the existing geotiff to the .png format
+        gdal.Translate(browse_temp_file,
+                       filename,
+                       # outputSRS="+proj=longlat +ellps=WGS84",
+                       # format='PNG',
+                       height=browse_image_height,
+                       width=browse_image_width,
+                       resampleAlg=resamp_algorithm)
+
+        gdal_ds = gdal.Open(browse_temp_file, gdal.GA_ReadOnly)
+
+    gdal_band = gdal_ds.GetRasterBand(1)
+    band_image = np.asarray(gdal_band.ReadAsArray(), dtype=np.float32)
+    is_valid = np.isfinite(band_image)
+    if alpha_channel is None:
+        alpha_channel = np.asarray(is_valid,
+                                   dtype=np.float32)
```

Comment:
```suggestion
    alpha_channel = np.asarray(is_valid,
                               dtype=np.float32)
```
The variable `alpha_channel` in this function is always `None` (see around line 409), so I do not see the necessity of the if statement here.
Same for the function `save_browse_imagery()`

Reply:
I fixed that for static layers. For imagery, that if statement is inside a for loop and we only need to assign values to alpha channel once.